includes/admin/discounts/class-discount-codes-table.php

```diff
@@ -189,55 +203,111 @@ protected function get_primary_column_name() {
 	 * @access public
 	 * @since 1.4
 	 *
-	 * @param array $item Contains all the data of the discount code
+	 * @param EDD_Discount $item Discount object.
 	 * @param string $column_name The name of the column
 	 *
 	 * @return string Column Name
 	 */
 	function column_default( $item, $column_name ) {
-		return $item[ $column_name ];
+		return property_exists( $item, $column_name ) ? $item->$column_name : '';
+	}
+
+	/**
+	 * This function renders the amount column.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @param EDD_Discount $item Data for the discount code.
+	 * @return string Formatted amount.
+	 */
+	public function column_amount( $item ) {
+		return edd_format_discount_rate( $item->type, $item->amount );
+    }
+
+	/**
+	 * This function renders the start column.
+	 *
+	 * @access public
+	 * @since 3.0
+	 *
+	 * @param EDD_Discount $item Discount object.
+	 * @return string Start  date
+	 */
+	function column_start_date( $item ) {
+		$start_date = $item->get_start_date();
+
+		if ( $start_date ) {
+			$display = date( 'F j, Y', strtotime( $start_date ) );
```

Comment:
Should we use `date_i18n` when displaying dates in the list table?

Reply:
Amended.